libvast/src/value_index.cpp

```diff
@@ -131,7 +131,7 @@ caf::error inspect(caf::deserializer& source, value_index_ptr& x) {
     x = nullptr;
     return caf::none;
   }
-  x = factory<value_index>::make(std::move(t));
+  x = factory<value_index>::make(std::move(t), options{});
```

Comment:
Maybe we should add a comment here:
```
// Persisted value indices shall not be parameterized at re-materialization
// time. Instead, a concrete index is responsible to persist its own metadata.
```

Reply:
I would really like to have this sort of guarantee, keeping the options only relevant to the factory. However, we have to support container types as well, which perform internal calls to the factory function and thus need to see the options. Consequently, the container indexes must get the options/settings in the constructor, and they must also store them as persistent state.

Reply:
As we figured out together at the HQ, there's a much bigger problem here: since the options can contain data that determines the concrete type of the hierarchy, they *must* be persisted as well. Only then the factory function will correctly default-construct the most derived type.